files/scripts/syncd.sh

```diff
@@ -163,11 +163,21 @@ stop() {
         debug "${TYPE} shutdown syncd process ..."
         /usr/bin/docker exec -i syncd$DEV /usr/bin/syncd_request_shutdown --${TYPE}
 
-        # wait until syncd quits gracefully
-        while docker top syncd$DEV | grep -q /usr/bin/syncd; do
+        # wait until syncd quits gracefully or force syncd to exit after 
+        # waiting for 20 seconds
+        start_in_secs=${SECONDS}
+        end_in_secs=${SECONDS}
+        timer_threshold=20
```

Comment:
why 20 seconds? what is the rationale?

Reply:
This threshold 20 seconds is what we discussed in our weekly meeting.

Reply:
there must be some rationale to set it to 20 seconds, why not 10 seconds? can you explain it?

Reply:
Actually when I tested the auto-restart feature of swss container, syncd container sometimes can be normally stopped and sometimes can not be stopped. The first thing I did is to profile how long this while loop will spin if syncd container can be normally stopped after swss container is stopped. The result is 5 seconds or 6 seconds. After that, in our meeting, we talked about to set a 20 seconds timer for this while loop before forcing syncd to exit. Please review the section of "Why I did it" of this PR for more details.

Sure, I can also try to set a 10 seconds timer to check whether it works or not.